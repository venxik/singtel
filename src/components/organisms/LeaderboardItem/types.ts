import { ViewProps } from '../../atoms';

export type LeaderboardItemProps = {
  text: string;
} & ViewProps;
