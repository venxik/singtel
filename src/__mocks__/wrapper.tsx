import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
