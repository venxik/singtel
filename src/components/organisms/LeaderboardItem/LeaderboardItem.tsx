import { ComicText, View, useThemeColor } from '../../atoms';
import React from 'react';
import { LeaderboardItemProps } from './types';
import { StyleSheet } from 'react-native';

const LeaderboardItem = (props: LeaderboardItemProps) => {
  const { style, lightColor, darkColor, text, ...otherProps } = props;

  const borderColor = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return (
    <View
      style={[
        {
          borderRadius: 6,
          justifyContent: 'center',
          width: '100%',
          height: 60,
          borderWidth: 1,
          borderColor,
        },
        style,
      ]}
      {...otherProps}>
      <ComicText testID="leaderboard-item-text" style={styles.text}>
        {text}
      </ComicText>
    </View>
  );
};

const styles = StyleSheet.create({
  text: { fontSize: 20, marginLeft: 20 },
});

export default React.memo(LeaderboardItem);
