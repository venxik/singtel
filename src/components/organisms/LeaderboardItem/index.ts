import LeaderboardItem from './LeaderboardItem';

export { LeaderboardItem };
