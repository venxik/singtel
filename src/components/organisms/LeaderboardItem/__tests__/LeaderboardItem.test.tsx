import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import LeaderboardItem from '../LeaderboardItem';

describe('LeaderboardItem', () => {
  it(`match snapshot`, () => {
    const tree = renderWithProviders(<LeaderboardItem text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`can have correct text`, () => {
    const { getByTestId } = renderWithProviders(
      <LeaderboardItem testID="category-item" text="test" />
    );
    expect(getByTestId('leaderboard-item-text').props.children).toEqual('test');
  });
});
