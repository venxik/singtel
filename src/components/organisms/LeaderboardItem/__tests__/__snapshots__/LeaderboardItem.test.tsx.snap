// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardItem match snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "borderColor": "#000",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 60,
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "fontSize": 20,
            "marginLeft": 20,
          },
          {
            "fontFamily": "comic",
          },
        ],
      ]
    }
    testID="leaderboard-item-text"
  >
    test
  </Text>
</View>
`;
